File: insertionSort/insertionSort.js

//  insertion sort
//  its by checking and swapping the adjacent elements 
// first iterate throught the elements 
//  then check whether the adjacent one is lesser then swap then repeat all the steps



let items = [18,-32,-11,6,68,20,-34]


for(i=0; i < items.length; i++){


    
    var temp = items[i]

    for(j= i-1; i>=0 && items[j] > temp; j--){
        items[j+1] = items[j]
    }

    items[j+1] = temp
}

console.log(items)

File: insertionSort/insertionSort.java

//  insertion sort
//  its by checking and swapping the adjacent elements 
// first iterate throught the elements 



public class insertionSort{


    public static void main(String args[]){

        int [] items = {18,-32,-110,6,-68,20,-34};

        for(int i=1; i< items.length; i++){

            int temp = items[i]; 
            int j = i-1;
            for( ; j >= 0 && items[j] > temp; j--){
                items[j+1] = items[j];
            }

            items[j+1] = temp;


        }

        for(int i=0; i< items.length; i++){
           System.out.println(items[i]);
        }
    }


}